src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs

```diff
@@ -201,14 +201,14 @@ private CSharpCompilation Rewrite(CSharpCompilation compilation)
 
         // Internal for unit testing
         internal CompilationResult GetCompilationFailedResult(
-            string relativePath,
+            RelativeFileInfo fileInfo,
             string compilationContent,
             string assemblyName,
             IEnumerable<Diagnostic> diagnostics)
         {
             var diagnosticGroups = diagnostics
                 .Where(IsError)
-                .GroupBy(diagnostic => GetFilePath(relativePath, diagnostic), StringComparer.Ordinal);
+                .GroupBy(diagnostic => GetFilePath(fileInfo.RelativePath, diagnostic), StringComparer.Ordinal);
```

Comment:
Hmm, is there any sort of security concern with showing full paths? Users obviously shouldn't show verbose diagnostic pages in production but that doesn't mean they wont.